tsl/src/continuous_aggs/create.c

```diff
@@ -1518,6 +1543,12 @@ mattablecolumninfo_addentry(MatTableColumnInfo *out, Node *input, int original_q
 				{
 					PRINT_MATCOLNAME(colbuf, "grp", original_query_resno, matcolno);
 					colname = colbuf;
+
+					/* for finalized form we skip those columns */
+					if (finalized)
+					{
+						*skip_adding = true;
+					}
 				}
```

Comment:
What's the assumption here? Is it :
```
Assert(tle->resname == NULL && tle->ressortgroupref > 0 && tle->resjunk == true);
```
Are we only aiming at group-by targets here?

Reply:
The finals form does not require those extra-group-by columns in the materialization hypertable

Reply:
I think I understand more now, there are some cases which we should document, and see if we're missing something if anything.

- `resname != NULL` means it's projected in our case
- `ressortgroupref > 0` means part of GROUP BY, which can be projected or not, depending on whether the column was deduplicated in the target list
- `resjunk == true` applies for GROUP BY columns that are not projected

So, in this scenario:
```
CREATE MATERIALIZED VIEW conditions_summary
    WITH (timescaledb.continuous, timescaledb.materialized_only=true)
AS SELECT
    time_bucket('1 days', time) AT TIME ZONE 'UTC',
    device_id,
    avg(temperature)
FROM conditions
GROUP BY device_id, time_bucket('1 days', time)  WITH NO DATA;
```
we get an index on `time_bucket()` regardless, and we get an index on `device_id` because it's deduplicated in the target-list and merged, as a member of both the GROUP BY and the projection.

However in this case:
```
CREATE MATERIALIZED VIEW conditions_summary
    WITH (timescaledb.continuous, timescaledb.materialized_only=true)
AS SELECT
    time_bucket('1 days', time) AT TIME ZONE 'UTC',
    device_id + 1,
    avg(temperature)
FROM conditions
GROUP BY device_id, time_bucket('1 days', time)  WITH NO DATA;
```
We never get the index on `device_id`.

Reply:
Awesome @mfundul ... your explanation here summarize everything.